### DISCUSSION TITLE
Dependence on This Integration

### DISCUSSION THREAD

**Discussion by mediacutlet** [tsightler/ring-mqtt#863] [General Discussion]
Now, that Ring's servers have sorted themselves out (referring to #861) I've had a moment to reflect on how dependent I've become on this integration.

I remember when I first installed this, I had no intention of making it a major backbone of my automation hierarchy.  Over time, I've become used to the apparent stability of this integration and rely on it more heavily than I'd like to admit.

That said, today's fire drill has me reconsidering some things and wondering if anyone has taken steps to move away from this integration or Ring Security in general. I use their retrofit system which takes advantage of all the existing sensors in the home which were installed in the 90s. This is a huge benefit to sticking with Ring for me. Doing some experimenting with ESPHome, I was able to adhoc the retrofit behavior with an ESP32. But I worry about the stability of going this route. Ring's Z-wave radio in has been rock solid.

I do have a separate Z-wave network running. Has anyone moved to manually configured Ring sensors?

I'd like to become less reliant on Ring but I also don't want to have to re-architect my entire setup. Curious to see how others keep dependence at an arm's length or have migrated away from this.

**Comment by tsightler** [comment]
I feel like this is a little offtopic for this discussion group.  I'm going to leave the discussion, but please remember that this discussion group is primarily for users leveraging the Ring integration, not looking to get away from it.  There are plenty of places to discuss homebrew sensors/alarm setups, etc. with lots of information out there.  Still, as long as we don't get into any Ring bashing, I'll let the thread play out.  Not that the OP did that, but I've found these threads can devolve pretty quickly as there's a fairly large subset of people that love to bash Ring for various decisions around features hidden behind paywall, so these topics make me nervous.

Personally, I built this integration 5+ years ago because I bought into Ring Alarm nearly at release, and I wanted to be able to leverage the existing alarm sensors rather than putting two sensors on every door, and having additional motion sensors.  This was a compromise for me since I generally prefer local control, but I also wanted a professionally monitored alarm system, which meant cloud connectivity, and there was always a chance of an outage, but, being an alarm, I suspected that Ring's would have to focus on reliability, while their doorbell/camera network was not quite so reliable, although the entire service level has improved over the years.  So far, it's been reliable enough, it's probably the 2nd or 3rd most reliable integration that I use, and probably the most reliable "cloud" integration.

However, I actually don't use Ring as my primary Z-wave hub, only for the things that are parts of the alarm subsystem.  Even though Ring supports locks, thermostats, switches, etc, I just connect those directly to Home Assistant via a Z-wave/Zigbee stick, vs sending them through Ring alarm.  I do have one lock connected to Ring, mostly for testing ring-mqtt with locks, same with thermostats, and a single switch, but everything critical to my home automation is local.

My local Z-wave stuff has also been rock solid stable, and I make heavy use of local automations, and it's really nice that they all work regardless of any outage.  I also have a fairly decent sized mesh of Zigbee devices, and they've also been fairly reliable, but not to the level of the Z-wave devices.

**Comment by mediacutlet** [comment]
Hearing your perspective @tsightler and how you use this integration is helpful. Perhaps I titled this thread too aggressively; overall I'm happy with Ring products and I'd prefer to keep things as they are. I understand their weaknesses and try to exploit their strengths to the fullest. 

I appreciate your insights and commend you for building a really great integration. I remember the day I discovered this how giddy I was and how impressed I was with your work. I still feel this way. I suppose I just got a little spooked today. **Thanks for all the time you've put into this!**

**Reply by tsightler** [reply]
Well, I'll never criticize anyone for wanting more local control and not being dependent on cloud services.  I really hope that Ring might eventually make their Z-wave network locally as a Thread border router, but that might be wishful thinking.

Also, I forgot one other thing, I wanted to use the sensors, but I also wanted to automate arming/disarming, which Ring didn't even offer at the time, and I believe they still only has fairly basic, non-conditional scheduling, while my legacy alarm system had a whole set of rules you could use.

**Comment by calisro** [comment]
It's definitely extremely reliable but it is also cloud.  I also wish it was local but Yes it's pretty much the backbone now of my HA in the same way.  I have all kinds of dependencies on the sensors and detectors to automate all kinds of other aspects of my HA system.  I do realize today that if it were offline for a few days, I would have issues with many automations but I think my future course of action will be to build into my existing setup ways of excluding the ring components and forcing them to work without the ring components in the event this happens again.  Personally, I don't see any issues being very long term but I do believe I need to plan for the possibility of this component to be offline for an extended period of time just in case.

Today my lights were turning off on me and wouldn't turn on, my kids were complaining and my wife was annoyed. :)  That's the major issue as Ring is currently being used as part of my occupancy automations.  Should be fairly easy though to build a bypass in so I can just bypass all of it with an input_boolean or even in a conditional automated fashion when the components are unavailable.

We learn we fix.  Thanks for this great addon.   This addon has been rock solid for years for me.

**Reply by tsightler** [reply]
Yeah, I definitely don't use this integration for anything that impactful.  I have some lighting automations, but only for ourdoor lighting, such as opening the front or back door will automatically turn on the front yard/back yard lights, but I also have Ring floodlights that will turn on individually with motion even if everything is down.  Yes, it's not as much light as turning on all the yard lights, but it's not zero.  Also, it's outside, so not something that would impact day-to-day that much.

I also have an "arrival" routine that recognizes the pattern of a person arriving, opening the garage door, opening the inside door, etc, that will automatically turn on lights based on time-of-day, which is really nice and my wife loves it, like the house is waking up to welcome her home.  But, I have failback, for example I have Ring sensors on the garage door and inside door, but I also have a Tailwind garage door opener which I've setup with local API access so I can trigger with that, and the lights are locally controlled switches, so there would at least be something. even if Ring was completely down.

I guess I would say I use this integration more as an additional supplement vs the core of anything or in places where it just adds a little extra, like using the keypad proximity sensor as an extra motion sensor in the entryway to turn the lights on as the hallway itself is kind of dark (I turn it on dim at night and bright during the day).

**Reply by mediacutlet** [reply]
You sure both of you don't live in my house? My setup is basically a hybrid of both of yours. Oddly pleasing to hear that you ended up with very similar use cases. Here are a few more since we're sharing.
- The evening "arm" and morning "disarm" have become important triggers. For instance, in the evening, when armed, an automation will test for occupancy in all rooms on the first floor and basement and slowly dim the lights off in those areas. (a variety of sensors help with this including mmw, power detection on tv outlet etc.)
- When the front door is opened, hush "person detected at the front door" announcements for a minute (Ring doorbell camera).
- Basement lights turn on when any door to the basement is opened. When the doors are "safe" again for a few minutes and motion in the basement is none, lights off.
-  Dashboards. Mobile, for instance, has conditional mushroom chips at the top including showing any doors/windows open. The wife particularly likes that one.

![image](https://github.com/tsightler/ring-mqtt/assets/26329687/6a077e0c-e5c2-4511-ac0c-b255266a2e1f)

I like the idea of creating redundancies and fallbacks so if this does go down, there is a backup in place; much better than being broken. Thanks for the suggestion. This would be prudent to do even without this reversed engineered integration and we were relying on all native Ring software; any cloud server can go down.

This concept has worked well with other systems like falling back to OpenWeather if my local weather measuring device fails.
